harness/determined/pytorch/_trainer.py

```diff
@@ -0,0 +1,246 @@
+
+    def configure_profiler(
+        self, sync_timings: bool, enabled: bool, begin_on_batch: int, end_after_batch: int
+    ) -> None:
+        if self._info is None:
+            raise ValueError("Determined profiler must be run on cluster.")
+
+        self._det_profiler = profiler.ProfilerAgent(
+            trial_id=str(self._info.trial.trial_id),
+            agent_id=self._info.agent_id,
+            master_url=self._info.master_url,
+            profiling_is_enabled=enabled,
+            global_rank=self._core.distributed.get_rank(),
+            local_rank=self._core.distributed.get_local_rank(),
+            begin_on_batch=begin_on_batch,
+            end_after_batch=end_after_batch,
+            sync_timings=sync_timings,
+        )
+
+    def fit(
+        self,
+        checkpoint_period: Optional[pytorch.TrainUnit] = None,
+        validation_period: Optional[pytorch.TrainUnit] = None,
+        max_length: Optional[pytorch.TrainUnit] = None,
+        reporting_period: pytorch.TrainUnit = pytorch.Batch(100),
+        checkpoint_policy: str = "best",
+        latest_checkpoint: Optional[str] = None,
+        step_zero_validation: bool = False,
+        test_mode: bool = False,
+    ) -> None:
+
+        # Set defaults.
+        if checkpoint_period is None:
+            checkpoint_period = pytorch.Batch(sys.maxsize)
+
+        if validation_period is None:
+            validation_period = pytorch.Batch(sys.maxsize)
+
+        if self._local_training:
+            if checkpoint_policy == "best":
+                logging.warning(
+                    "checkpoint_policy='best' is not supported in local training mode. "
+                    "Falling back to 'all'."
+                )
+                checkpoint_policy = "all"
+            if max_length is None:
+                raise ValueError("max_length must be defined in local training mode.")
+
+            if not isinstance(max_length.value, int):
+                raise TypeError("max_length must be configured in TrainUnit(int) types.")
+
+            if self._det_profiler:
+                logging.warning("Determined profiler will be ignored in local training mode.")
+
+            smaller_is_better = True
+            searcher_metric_name = None
+            steps_completed = 0
+            global_batch_size = None
+        else:
+            if test_mode:
+                raise ValueError("test_mode is only supported in local training mode.")
+
+            if max_length is not None:
+                logging.warning(
+                    "max_length is ignored when training on-cluster. Please configure the searcher instead."
+                )
+
+            if latest_checkpoint is None:
+                logging.warning(
+                    "latest_checkpoint has not been configured. Pause/resume training will not be able to continue "
+                    "from latest checkpoint. Did you mean to configure 'info.trial.latest_checkpoint'?"
```

Comment:
rephrase, since you can't "configure info.trial.latest_checkpoint".  Maybe, "did you mean to set fit(latest_checkpoint=info.trial.latest_checkpoint)"?